src/test/state_test.rs

```diff
@@ -0,0 +1,122 @@
+use indexmap::IndexMap;
+use serde_json::json;
+
+use crate::deprecated_contract_class::EntryPointOffset;
+
+#[test]
+fn entry_point_offset_from_json_str() {
+    let data = r#"
+        {
+            "offset_1":  2,
+            "offset_2": "0x7b"
+        }"#;
+    let offsets: IndexMap<String, EntryPointOffset> = serde_json::from_str(data).unwrap();
+
+    assert_eq!(EntryPointOffset(2), offsets["offset_1"]);
+    assert_eq!(EntryPointOffset(123), offsets["offset_2"]);
+}
+
+#[test]
+fn entry_point_offset_into_json_str() {
+    let offset = EntryPointOffset(123);
+    assert_eq!(json!(offset), json!(format!("{:#x}", offset.0)));
+}
+
+#[cfg(test)]
+mod thin_state_diff_tests {
```

Comment:
Oh some tests are there, but I have the impressions that not every struct that has a manual impl, also has a test